ui-modules/app-inspector/app/views/main/inspect/summary/summary.controller.js

```diff
@@ -100,15 +101,10 @@ export function summaryController($scope, $state, $stateParams, $q, $http, $http
 
             // TODO: ideally move this to a $watch block
             if (vm.config && vm.configResolved && vm.configInfo) {
-                vm.configItems = Object.entries(vm.showResolvedConfig ? vm.configResolved : vm.config)
-                    .map(([key, value]) => ({
-                        key,
-                        value,
-                        // marking as unsafe if the field name looks sensitive
-                        // and the unresolved value does *not* come from a secure external source
-                        isUnsafe: isSensitiveFieldName(key.trim()) &&
-                            !vm.config[key].toString().startsWith('$brooklyn:'),
-                    }));
+                vm.configItems = vm.showResolvedConfig ? vm.configResolved : vm.config;
+                vm.configItemsUnsafeMap = mapValues(vm.configItems, (value, key) =>
+                    isSensitiveFieldName(key.trim()) && !vm.config[key].toString().startsWith('$brooklyn:')
```

Comment:
will this fail if not set?

i will test and fix

Reply:
ie `vm.config[key]` can be null can't it?

also do we need `mapValues` ... can't we just use Object.keys(vm.configItems).forEach(key => ...)`?

finally on reflection i think it would be cleaner to pass a function eg `isSensitive(key, value)` to the widget, rather than a map.

if no objections i'll apply this and merge.

Reply:
the lodash `mapValues` (like most of its functions) is very resistant to "bad" values, so worse scenario we'll have an empty object for the config-sensor table, meaning no warning icons at all but everything else should be fine.

![image](https://user-images.githubusercontent.com/8679925/133661927-43252290-6079-4950-98f2-ddb00a3d8684.png)

Reply:
i knew i'd learn something by asking!  ta